util/pingpong.c

```diff
@@ -0,0 +1,2111 @@
+	struct fid_ep *ep;
+	struct fid_cq *txcq, *rxcq;
+	struct fid_mr *mr;
+	struct fid_av *av;
+	struct fid_eq *eq;
+
+	struct fid_mr no_mr;
+	struct fi_context tx_ctx, rx_ctx;
+	struct fi_context *ctx_arr;
+	uint64_t remote_cq_data;
+
+	uint64_t tx_seq, rx_seq, tx_cq_cntr, rx_cq_cntr;
+	//int ft_skip_mr;
+	int ft_parent_proc;
+	pid_t ft_child_pid;
+
+	fi_addr_t remote_fi_addr;
+	void *buf, *tx_buf, *rx_buf;
+	size_t buf_size, tx_size, rx_size;
+	int rx_fd, tx_fd;
+	int data_default_port;
+	char data_port[8];
+
+	char test_name[50];
+	int timeout;
+	struct timespec start, end;
+
+	struct fi_av_attr av_attr;
+	struct fi_eq_attr eq_attr;
+	struct fi_cq_attr cq_attr;
+	struct ft_opts opts;
+
+	long cnt_ack_msg;
+
+	int ctrl_port;
+	int ctrl_listenfd;
+	int ctrl_connfd;
+	struct sockaddr_in ctrl_addr;
+	char ctrl_buf[CTRL_BUF_LEN + 1];
+};
+
+#define FT_ENABLE_ALL		(~0)
+#define FT_DEFAULT_SIZE		(1 << 0)
+
+struct test_size_param {
+	int size;
+	int enable_flags;
+};
+
+struct test_size_param test_size[] = {
+	{ 1 <<  1, 0 }, { (1 <<  1) + (1 <<  0), 0 },
+	{ 1 <<  2, 0 }, { (1 <<  2) + (1 <<  1), 0 },
+	{ 1 <<  3, 0 }, { (1 <<  3) + (1 <<  2), 0 },
+	{ 1 <<  4, 0 }, { (1 <<  4) + (1 <<  3), 0 },
+	{ 1 <<  5, 0 }, { (1 <<  5) + (1 <<  4), 0 },
+	{ 1 <<  6, FT_DEFAULT_SIZE }, { (1 <<  6) + (1 <<  5), 0 },
+	{ 1 <<  7, 0 }, { (1 <<  7) + (1 <<  6), 0 },
+	{ 1 <<  8, FT_DEFAULT_SIZE }, { (1 <<  8) + (1 <<  7), 0 },
+	{ 1 <<  9, 0 }, { (1 <<  9) + (1 <<  8), 0 },
+	{ 1 << 10, FT_DEFAULT_SIZE }, { (1 << 10) + (1 <<  9), 0 },
+	{ 1 << 11, 0 }, { (1 << 11) + (1 << 10), 0 },
+	{ 1 << 12, FT_DEFAULT_SIZE }, { (1 << 12) + (1 << 11), 0 },
+	{ 1 << 13, FT_DEFAULT_SIZE }, { (1 << 13) + (1 << 12), FT_DEFAULT_SIZE },
+	{ 1 << 14, FT_DEFAULT_SIZE }, { (1 << 14) + (1 << 13), 0 },
+	{ 1 << 15, 0 }, { (1 << 15) + (1 << 14), 0 },
+	{ 1 << 16, 0}, { (1 << 16) + (1 << 15), 0 },
+	{ 1 << 17, 0 }, { (1 << 17) + (1 << 16), 0 },
+	{ 1 << 18, 0 }, { (1 << 18) + (1 << 17), 0 },
+	{ 1 << 19, 0 }, { (1 << 19) + (1 << 18), 0 },
+	{ 1 << 20, 0}, { (1 << 20) + (1 << 19), 0 },
+	{ 1 << 21, 0}, { (1 << 21) + (1 << 20), 0 },
+	{ 1 << 22, 0}, { (1 << 22) + (1 << 21), 0 },
+	{ 1 << 23, 0 },
+};
+
+const unsigned int test_cnt = (sizeof test_size / sizeof test_size[0]);
+
+static const char integ_alphabet[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+static const int integ_alphabet_length = (sizeof(integ_alphabet)/sizeof(*integ_alphabet)) - 1;
+
+#define TEST_CNT test_cnt
+#define INTEG_SEED 7
+#define FT_BENCHMARK_MAX_MSG_SIZE (test_size[TEST_CNT - 1].size)
+
+
+/*******************************************************************************************/
+/*                                       FT proto                                          */
+/*******************************************************************************************/
+
+char *ep_name(int ep_type);
+void ft_banner_info(struct ct_pingpong *ct);
+void ft_pingpong_pusage(char *name, char *desc);
```

Comment:
I think that I removed and added too many things so that it would be a pain to merge it back with the functions/variables/macros of fabtests.
So maybe `pp_` is a good idea.